### PR TITLE
[ML][Pipelines] Support output setting copy in SDK

### PR DESCRIPTION
**Pull request by D-W-**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

In this PR, SDK will copy node level output settings to pipeline level when node output promoted to pipeline level.
The following settings will be copied:
- Type
- Mode
- Path

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_pipeline_component_builder.py [hunk 1]
```diff
@@ -271,22 +272,24 @@ def _map_type(_meta):
...
+                type=_map_type(meta),
+                description=meta.description,
+                mode=meta.mode,
+                is_control=meta.is_control
             )
             pipeline_output = PipelineOutput(
                 port_name=key,
                 data=None,
+                # This is a hack: when parallel_for node outputs promoted to pipeline level,
+                # pipeline outputs type should be same with node levle.
```
Comment (zhengfeiwang) [inline sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_pipeline_component_builder.py:284]: typo here: levle => level

Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_pipeline_component_builder.py:284]: and is this comment for change in this PR? seems like comment for previous change?

Comment (D-W-) [reply sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_pipeline_component_builder.py:284]: Good catch!

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_util.py [hunk 2]
```diff
@@ -458,3 +459,15 @@ def get_type_from_spec(data: dict, *, valid_keys: Iterable[str]) -> str:
...
     return extract_label(_type)[0]
+
+
+def copy_output_setting(source: "NodeOutput", target: "NodeOutput"):
+    """Copy node output setting from source to target.
+    Currently only path, name, version will be copied."""
+    # pylint: disable=protected-access
+    from azure.ai.ml.entities._job.pipeline._io import NodeOutput
+
+    if not isinstance(source, NodeOutput):
```
Comment (zhengfeiwang) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_util.py:470]: may add comment for this exception?

File: sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_pipeline_component_builder.py (hunk 1 above)
Comment (D-W-) [reply sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_pipeline_component_builder.py:284]: It's newly added, I can't remember why we need to add meta here.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_util.py (hunk 2 above)
Comment (D-W-) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_util.py:470]: Updated interface

File: sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_pipeline_component_builder.py (hunk 1 above)
Comment (D-W-) [reply sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_pipeline_component_builder.py]: Updated

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/_io/mixin.py [hunk 3]
```diff
@@ -320,18 +321,53 @@ def _build_input(self, name, meta: Input, data) -> "PipelineInput":
...
+            raise ValidationException(
+                message=message.format(type(data), name, data),
+                no_personal_data_message=message,
+                target=ErrorTarget.PIPELINE,
+            )
+        return output_val
+
+    def _build_pipeline_outputs_dict(self, outputs: Dict[str, Union[Output, NodeOutput]]) -> OutputsAttrDict:
+        """Build an output attribute dict without output definition metadata.
+        For pipeline outputs, it's setting should be copied from node level outputs.
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/_io/mixin.py:361]: it's -> its

File: sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py [hunk 4]
```diff
@@ -2666,6 +2666,80 @@ def test_dsl_pipeline_with_data_transfer_copy_2urifolder(self, client: MLClient)
...
+            return node1.outputs
+
+        pipeline_job = outer_pipeline()
+        pipeline_job_dict = pipeline_job._to_dict()
+        assert pipeline_job_dict["outputs"] == {
+            'component_out_path': {'path': 'azureml://datastores/workspaceblobstore/paths/outputs/1'}
+        }
+        pipeline_component = pipeline_job.jobs["node1"].component
+        pipeline_component_dict = pipeline_component._to_dict()
+        assert pipeline_component_dict["outputs"] == {'component_out_path': {'type': 'uri_folder'}}
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py:2723]: maybe use a uri_file test case here, so we know this is not a default value here

Comment (wangchao1230) [reply sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py:2723]: I saw there is ut covering this below

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**